packages/runtime/runtime-utils/src/garbageCollector.ts

```diff
@@ -0,0 +1,82 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryLogger } from "@fluidframework/common-definitions";
+import { IGraphNode } from "@fluidframework/runtime-definitions";
+
+export interface IGCResult {
+    referencedNodes: IGraphNode[];
+    deletedNodes: IGraphNode[];
+}
+
+export interface IGarbageCollector {
+    /**
+     * Runs GC on the given reference graph. startingIds are ids of nodes from which the GC algorithm starts
+     * and searches for other referenced nodes.
+     */
+    runGC(referenceGraph: IGraphNode[], startingIds: string[]): IGCResult;
```

Comment:
should startingIds be rootIds? It's not very clear what "starting" means in this context